app/derivative-services/jp2_derivative_service.rb

```diff
@@ -180,4 +180,17 @@ def cleanup_derivative_metadata(derivatives:)
         buffered_persister.save(change_set: updated_change_set)
       end
     end
+
+    def storage_adapter
+      @storage_adapter ||= Valkyrie::StorageAdapter.find(:derivatives)
+    end
+
+    # Updates error message property on the original file.
+    def update_error_message(message:)
```

Comment:
Is there somewhere we can standardize this method? It's copied/pasted in every derivative service yeah?

Reply:
I agree, it needs some DRYing out. I was conflicted about the best approach -  mixin module probably. However, it's not included in every derivative service (JP2, Image, Vector, Raster only). There are a handful of other methods with the same issue. `cleanup_derivative_metadata`, `file_object`, and `storage_adapter` for example.

Reply:
In the event that we do decide to DRY here, I would hope to explore inheritance. I don't think a mixin would be the right choice here; it seems like these are specific to derivatives creation.